gulpfile.js

```diff
@@ -73,15 +73,24 @@ gulp.task('test.unit', ['test.compile'], function(done) {
     done();
     return;
   }
-  return gulp.src('build/test/**/*.js').pipe(mocha({timeout: 500}));
+  return gulp.src(['build/test/**/*.js', '!build/test/**/e2e*.js']).pipe(mocha({timeout: 1000}));
 });
 
-gulp.task('test', ['test.unit', 'test.check-format']);
+gulp.task('test.e2e', ['test.compile'], function(done) {
+  if (hasError) {
```

Comment:
Yeah, gulp has no systematic way to deal with errors, and surprisingly it's still better than all the other tools in that regard...